src/Cardano/Shell/Features/Logging.hs

```diff
@@ -57,33 +44,35 @@ testRotationParameters = RotationParameters
 -- the functions effects and constraining the user (programmer) of those function to use specific effects in them.
 -- https://github.com/input-output-hk/cardano-sl/blob/develop/util/src/Pos/Util/Log/LogSafe.hs
 data LoggingLayer = LoggingLayer
-    { iolLogDebug :: forall m. (MonadIO m) => Text -> m ()
-    , iolLogInfo  :: forall m. (MonadIO m) => Text -> m ()
-    , iolNonIo    :: forall m. (MonadThrow m) => m ()
-    }
-
-testLoggingLayer :: LoggingLayer
-testLoggingLayer = LoggingLayer
-    { iolLogDebug               = liftIO . putTextLn
-    , iolLogInfo                = liftIO . putTextLn
-    , iolNonIo                  = pure ()
+    { startTrace  :: forall m. (MonadIO m) => Trace m
+    , logDebug    :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , logInfo     :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , logNotice   :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , logWarning  :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , logError    :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , appendName  :: forall m. (MonadIO m) => Text -> Trace m -> m (Trace m)
+    , mockNonIO   :: forall m. (MonadThrow m) => m ()
```

Comment:
That is just an example of how a function _would look if it had a different constraint_. You don't have to add it to your interface. You can remove that function.